### PR TITLE
Swagger docs

### LINKED ISSUE #4: Document the endpoints using OpenAPI 3.0
Document the API endpoints using OpenAPI 3.0 so that it can generate Swagger (or similar) documentation.

views:
- [x] product
- [x] variety
- [x] project
- [x] activity
- [x] account
- [x] application
- [x] notification

Models:
- [x] product
- [x] variety
- [x] project
- [x] activity
- [x] account
- [x] application
- [x] notification
- [ ] Separate schemas for request and response (or use `readOnly` and `writeOnly`)

Security:
- [ ] Specify permissions (admin, user, project moderator, project creator)

other:
- [ ] more descriptive response descriptions and summaries
- [ ] use `operationId`s that coincide with the function name in the Flask code
- [ ] use descriptions from the endpoints' docstring
- [ ] use `x-tagGroups` for Redoc grouping
- [ ] document the different responses for the same response code (using `oneOf`. Ex: multiple possible error messages for the code 400)
- [x] Add `externalDocs` that point to Notion pages

### PR DESCRIPTION
**Pull request by aabounegm**
Closes #4

### CODE REVIEW

File: swagger.yaml [hunk 1]
```diff
@@ -15,9 +15,9 @@ externalDocs:
   url: https://www.notion.so/innopoints/
 
 servers:
+- url: https://ipts.innopolis.university/api/v1/
 - url: https://innopoints-backend.herokuapp.com/api/v1/
```
Comment (illright) [inline swagger.yaml]: This one's no longer relevant

File: swagger.yaml [hunk 2]
```diff
@@ -63,21 +63,48 @@ paths:
...
+          description: Maximum price of products
+          in: query
+          schema:
+            type: integer
+            default: null
+            nullable: true
+            minimum: 0
+          example: 1337
+        - name: excluded_colors
+          description: Which colors (in hex form) to filter from the returned products. `null` represents colorless
```
Comment (illright) [inline swagger.yaml]: Then it's not really `null` that represents colorless, it's the null byte. I understand that the null byte is a way of encoding a null in query strings, but others may not

File: swagger.yaml [hunk 3]
```diff
@@ -51,7 +51,7 @@ paths:
             default: 1
             minimum: 1
         - name: limit
-          description: Maximum number of elemenets
+          description: Maximum number of products
```
Comment (illright) [inline swagger.yaml]: I'd also add that it's the maximum number of products **per page**

File: swagger.yaml [hunk 4]
```diff
@@ -63,21 +63,48 @@ paths:
...
+          description: Which colors (in hex form) to filter from the returned products. `null` represents colorless
+          in: query
+          schema:
+            type: array
+            items:
+              type: string
+              nullable: true
+            default: []
+          example: ["FF0000", "\0"]
       responses:
```
Comment (illright) [inline swagger.yaml:110]: Can't comment on the line 134, but technically we have periods at the end of all error messages, including `Bad query parameters.`

File: swagger.yaml (hunk 1 above)
Comment (aabounegm) [reply swagger.yaml]: It is still authorized to access the SSO, so I thought we might as well leave it. Also, wasn't it set up to automatically pull from master?

File: swagger.yaml (hunk 2 above)
Comment (aabounegm) [reply swagger.yaml]: This wasn't intended. The description simply wasn't updated after replacing `null` with the null byte

File: swagger.yaml (hunk 4 above)
Comment (aabounegm) [reply swagger.yaml:110]: Actually, the error schema is something that should be structured well, and is currently inconsistent throughout the docs

File: swagger.yaml (hunk 1 above)
Comment (illright) [reply swagger.yaml]: I don't think it's still authorized. Besides, I brought it down months ago anyway, really no use in keeping it

**Pull request merged**